### PR TITLE
feat(capacity): add replenishable traits

### LINKED ISSUE #895: [Capacity] interface for replenishing capacity
### Description:
Implement interfaces in capacity pallet that allows checking and replenishing capacity.

```
traits Nontransferable {
  type Balance;

  /// The available Capacity for an MSA account.
  fn balance(msa_id: MessageSourceId) -> Result<Balance, DispatchError>;

  /// Reduce the available Capacity of an MSA account.
  fn deduct(msa_id: MessageSourceId, amount: Balance) -> Result<Balance, DispatchError>;

  /// Increase the available Capacity for an MSA account.
  fn deposit(msa_id: MessageSourceId, amount: Balance) -> Result<Balance, DispatchError>;
}
```

```
trait Replenishable {
  type Balance;

  /// Replenish an MSA's Capacity by an amount.
  fn replenish_by_amount(msa_id: MessageSourceId, amount: Balance) -> Result<Balance, DispatchError> {};

  /// Replenish all Capacity balance for an MSA.
  fn replenish_all_for_account(msa_id: MessageSourceId) -> Result<Balance, DispatchError>;

  /// Checks if an account can be replenished.
  fn can_replenish(msa_id: MessageSourceId) -> bool;
}
```

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Add nontransferable and replenishable traits
to Capacity.

Closes #895

### CODE REVIEW

**Comment by enddynayn** [comment]
This PR is a continuation of the following PR: https://github.com/LibertyDSNP/frequency/pull/946 
I cherry-picked the commit as an alternative to rebasing.

File: pallets/capacity/src/helpers_tests.rs [hunk 1]
```diff
@@ -265,3 +265,240 @@ fn calculate_capacity_reduction_determines_the_correct_capacity_reduction_amount
...
+		let amount = BalanceOf::<Test>::from(10u32);
+		assert_noop!(
+			Capacity::deduct(target_msa_id, amount),
+			Error::<Test>::TargetCapacityNotFound
+		);
+	});
+}
+
+#[test]
+fn impl_deduct_errors_insufficien_balance() {
```
Comment (enddynayn) [inline pallets/capacity/src/helpers_tests.rs:335]: https://github.com/LibertyDSNP/frequency/pull/946#discussion_r1084600005

File: pallets/capacity/src/helpers_tests.rs [hunk 2]
```diff
@@ -265,3 +265,240 @@ fn calculate_capacity_reduction_determines_the_correct_capacity_reduction_amount
...
+		let amount = BalanceOf::<Test>::from(10u32);
+		assert_noop!(
+			Capacity::deposit(target_msa_id, amount),
+			Error::<Test>::TargetCapacityNotFound
+		);
+	});
+}
+
+#[test]
+fn impl_deposit_errors_increase_exceeds_available() {
```
Comment (enddynayn) [inline pallets/capacity/src/helpers_tests.rs:393]: https://github.com/LibertyDSNP/frequency/pull/946#discussion_r1084605549

File: pallets/capacity/src/helpers_tests.rs [hunk 3]
```diff
@@ -265,3 +265,240 @@ fn calculate_capacity_reduction_determines_the_correct_capacity_reduction_amount
...
+			1u32,
+		);
+
+		assert_noop!(
+			Capacity::deduct(target_msa_id, 20u32.into()),
+			Error::<Test>::InsufficientBalance
+		);
+
+		let mut capacity_details =
+			CapacityDetails::<BalanceOf<Test>, <Test as Config>::EpochNumber>::default();
```
Comment (enddynayn) [inline pallets/capacity/src/helpers_tests.rs:353]: https://github.com/LibertyDSNP/frequency/pull/946/files#r1084601160

Comment (enddynayn) [reply pallets/capacity/src/helpers_tests.rs:353]: yes

**Comment by enddynayn** [comment]
@johnmela I removed the lock file changes.

https://github.com/LibertyDSNP/frequency/pull/946#discussion_r1084374718

**Review by shannonwells — APPROVED** [review]
🚢 it!

**Review by johnmela — APPROVED** [review]

**Pull request merged**